server/controllers/tasks/lib/get_new_tasks.js

```diff
@@ -15,21 +17,87 @@ module.exports = entity => existingTasks => {
     if (newSuggestions.length <= 0) return []
     const { labels: worksLabels } = suspectWorksData
     return Promise.all(newSuggestions.map(addOccurrencesToSuggestion(suspectWorksData)))
-    .then(evaluateSuggestions(entity, worksLabels))
-    .then(filterOutExistingTasks(existingTasks))
-    .then(buildTaskObjects(suspectUri))
+    .then(filterOrMergeSuggestions(entity, worksLabels))
+    .then(filterNewTasks(existingTasks))
   })
 }
 
-const buildTaskObjects = suspectUri => suggestions => suggestions.map(suggestion => ({
-  type: 'deduplicate',
-  suspectUri,
-  suggestionUri: suggestion.uri,
-  lexicalScore: suggestion._score,
-  externalSourcesOccurrences: suggestion.occurrences
-}))
+const filterOrMergeSuggestions = (suspect, workLabels) => suggestions => {
+  const suspectTerms = getEntityNormalizedTerms(suspect)
+  // Do not automerge if author name is in work title
+  // as it confuses occurences found wikipedia pages
+  if (authorNameInWorkTitles(suspectTerms, workLabels)) return suggestions
+  const sourcedSuggestions = filterSourced(suggestions)
+  if (sourcedSuggestions.length === 0) return suggestions
+  if (sourcedSuggestions.length > 1) return sourcedSuggestions
+  return automerge(suspect.uri, sourcedSuggestions[0])
+}
 
-const filterOutExistingTasks = existingTasks => suggestions => {
+const filterNewTasks = existingTasks => suggestions => {
   const existingTasksUris = _.map(existingTasks, 'suggestionUri')
   return suggestions.filter(suggestion => !existingTasksUris.includes(suggestion.uri))
 }
+
+const authorNameInWorkTitles = (authorTerms, workLabels) => {
+  for (const authorLabel of authorTerms) {
+    for (const workLabel of workLabels) {
+      return workLabel.match(authorLabel)
```

Comment:
this makes the loop return the match of the first `authorLabel` with the first `workLabel` and nothing else

```suggestion
      if (workLabel.match(authorLabel) != null) then return true
```

Reply:
[documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) says:

> `.match` returns an Array or null if no matches are found.

since arrays are truthy, this is what we need here. I mean that even when .match does not return a boolean, the if where this function is consumed understands it this way.

this PR does not modify this function, [its the same as before](https://github.com/inventaire/inventaire/blob/master/server/controllers/tasks/lib/evaluate_suggestions.js#L19)

Reply:
it's not the same as before, there was an `if` before (I shouldn't have added the `!= null` part in the suggestion above, as it's indeed irrelevant)

Reply:
[forcing myself to not continue this conversation :) ]